core/commands/files/files.go

```diff
@@ -471,7 +473,12 @@ Example:
 			res.SetError(err, cmds.ErrNormal)
 			return
 		}
-		defer fi.Close()
+
+		if noflush {
+			defer fi.Sync()
+		} else {
+			defer fi.Close()
+		}
```

Comment:
so this means that the `files/` dag is not consistent between calls? i.e. "flush" means "bubble up the change to the root"?

- what happens if the user never calls write again?
- what happens if program crashes before user can call write again?

the state of the dag should be recoverable.

maybe what we want is log journaled writes, so they're safe even without flushing the dag.